src/pages/workspaces/WorkspacesList/RenderedWorkspaces.ts

```diff
@@ -0,0 +1,311 @@
+import _ from 'lodash/fp';
+import { ReactNode, useContext, useState } from 'react';
+import { div, h, span } from 'react-hyperscript-helpers';
+import { AutoSizer } from 'react-virtualized';
+import { CloudProviderIcon } from 'src/components/CloudProviderIcon';
+import { Link } from 'src/components/common';
+import { FirstParagraphMarkdownViewer } from 'src/components/markdown';
+import { FlexTable, HeaderRenderer } from 'src/components/table';
+import TooltipTrigger from 'src/components/TooltipTrigger';
+import { WorkspaceStarControl } from 'src/components/WorkspaceStarControl';
+import { WorkspaceSubmissionStatusIcon } from 'src/components/WorkspaceSubmissionStatusIcon';
+import { Ajax } from 'src/libs/ajax';
+import colors from 'src/libs/colors';
+import Events, { extractWorkspaceDetails } from 'src/libs/events';
+import { getLink } from 'src/libs/nav';
+import { useStore } from 'src/libs/react-utils';
+import { AuthState, authStore } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import * as Utils from 'src/libs/utils';
+import {
+  canRead,
+  getCloudProviderFromWorkspace,
+  workspaceAccessLevels,
+  WorkspaceInfo,
+  WorkspaceWrapper as Workspace,
+} from 'src/libs/workspace-utils';
+import WorkspaceMenu from 'src/pages/workspaces/workspace/WorkspaceMenu';
+import { workspaceSubmissionStatus } from 'src/pages/workspaces/WorkspacesList/useWorkspacesWithSubmissionStats';
+import { WorkspaceUserActionsContext } from 'src/pages/workspaces/WorkspacesList/WorkspaceUserActions';
+
+// This is actually the sort type from the FlexTable component
+// When that component is converted to typescript, we should use that instead
+interface WorkspaceSort {
+  field: keyof WorkspaceInfo | keyof Workspace;
+  direction: 'desc' | 'asc';
+}
+
+const styles = {
+  tableCellContainer: {
+    height: '100%',
+    padding: '0.5rem 0',
+    paddingRight: '2rem',
+    borderTop: `1px solid ${colors.light()}`,
+  },
+  tableCellContent: {
+    height: '50%',
+    display: 'flex',
+    alignItems: 'center',
+  },
+};
```

Comment:
I like typing styles objects like this:

```suggestion
} as const satisfies Record<string, CSSProperties>;
```

It preserves the exact keys in the type while enforcing that all style properties are valid.